### PR TITLE
Handle error codes from node-influx client and propagate back to flow

### PR DESCRIPTION
**Pull request by malcolmwhat**
The error codes were not available in the error part of the `msg` since the `node-influx` cli has these defined inside their `err.res` which is not included in any `catch` nodes (from what I could tell).

I added the `statusCode` to the `msg` object in the catch of the three nodes so that users can handle the error codes in their flows.

I also bumped the version of lodash for #45

### CODE REVIEW

**Comment by mblackstock** [comment]
Thanks for this!  It looks like node-red is creating a new error object with only certain fields from the original error from the node.  To add our own error information specific to the node, I think it would be better to add it to an 'error' field in the message.  Node-RED then copies this over to an '_error' field -- I think this indicates these are node-specific fields that users can look at if they want node-specific error fields.

That said, I'm not sure why users need only the error.res.statusCode field when it exists, nor why we should add one if it isn't there.  Can users get what they need from the error message?  Are there other fields we may want to expose to the client from the RequestError thrown by the library?  Should we change the field to 'errorCode' or just 'code'?  What do you think?

**Comment by malcolmwhat** [comment]
No problem!

In the setup below I had a look at what comes out from when you raise `node.error` on the lines near what I change. Basically what it seems to be is that only the top-level error information gets copied into this `error` object (I added the statusCode here). The error code is in the message, but as you can see it's not pretty or very practical to deal with. The `.error` property actually seems to contain mostly details about the actual error as it pertains to `node-red`, and less so with the thrown error from the `node-influx` library.
image

```
{
   "_msgid":"2f796b6a.3f23d4",
   "topic":"",
   "payload":1567607322230,
   "statusCode":400,
   "error":{
      "message":"Error: A 400 Bad Request error occurred: {\"error\":\"error parsing query: only ORDER BY time supported at this time\"}\n",
      "source":{
         "id":"fe9d378a.fdd1e8",
         "type":"influxdb in",
         "name":"Test Direct",
         "count":1
      },
      "stack":"Error: A 400 Bad Request error occurred: {\"error\":\"error parsing query: only ORDER BY time supported at this time\"}\n\n    at IncomingMessage.Create.res.on (C:\\***\\node-red\\node_modules\\influx\\lib\\src\\pool.js:49:38)\n    at IncomingMessage.emit (events.js:203:15)\n    at endReadableNT (_stream_readable.js:1129:12)\n    at process._tickCallback (internal/process/next_tick.js:63:19)"
   }
}
```


The reasoning behind using `statusCode` was to follow the form of the `http-request` node.
> **payload string | object | buffer**
    The body of the response. The node can be configured to return the body as a string, attempt to parse it as a JSON string or leave it as a binary buffer.
**statusCode number**
    The status code of the response, or the error code if the request could not be completed.
[...] 

The influx API (which `node-influx` uses has a defined set of [status codes](https://docs.influxdata.com/influxdb/v1.7/tools/api/#status-codes-and-responses-1) that can be referenced by the user (i.e. me) to determine what has happened and what they can do next about it. The reasoning behind adding the `503` was that if there is no actual connection (i.e. a network error or the actual influx service fails) then the `statusCode` will be undefined but really this should be a [service unavailable 503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) error. I don't see a good way to do this in the node-influx library or in the user side of things. The way node-influx has sent the error is fine, we just need to propagate it up to the user in the catch.

> Are there other fields we may want to expose to the client from the RequestError thrown by the library?

Very likely, but this PR serves a pretty distinct purpose, and doing a more comprehensive coverage of the `node-influx` docs or even mapping their entire error object into the messages would be a lot more work than I've put in here 😝 I see this as a baby step

File: influxdb.js [hunk 1]
```diff
@@ -135,6 +135,11 @@ module.exports = function(RED) {
                 }
 
                 client.writePoints(points, writeOptions).catch(function(err) {
+                    if (err.res === undefined){
```
Comment (mblackstock) [inline influxdb.js]: Rather than adding a statusCode to a message when an exception occurs, can we simply add the error to the message like this?

```
msg.error = err;
node.error(err, msg);
```

This way the user can dig through the ResultError or RequestError by accessing the _error field in the message.  Not sure it will work though.  We may have to copy the fields that we want into a new object.

**Comment by mblackstock** [comment]
OK.  I don't think the way errors are handled here by the node and library is analogous to the http-request node.  My thinking is we can bubble up node-specific error information from the library in the _error field somehow and add new fields later if needed as I mentioned in the comment above.

File: influxdb.js (hunk 1 above)
Comment (malcolmwhat) [reply influxdb.js]: image
I tried that, weirdly though I don't think it worked. 

Below in the 400 you can see that my new code has added it to the msg, but not the err (a sanity check to make sure the write was working) and then you can see in the 503 there is nothing at all. I'm not sure where this _error value is, but the msg.error value seems to be populated by err, it's just not clear how this happens.

Forced a 400 with a bad query:
```
object
_msgid: "df7d6275.1c17e"
topic: ""
payload: 1567619174488
statusCode: 400
error: object
message: "Error: A 400 Bad Request error occurred: {"error":"error parsing query: only ORDER BY time supported at this time"}↵"
source: object
id: "fe9d378a.fdd1e8"
type: "influxdb in"
name: "Test Direct"
count: 1
stack: "Error: A 400 Bad Request error occurred: {"error":"error parsing query: only ORDER BY time supported at this time"}↵↵ at IncomingMessage.Create.res.on (***\node-red\node_modules\influx\lib\src\pool.js:49:38)↵ at IncomingMessage.emit (events.js:203:15)↵ at endReadableNT (_stream_readable.js:1129:12)↵ at process._tickCallback (internal/process/next_tick.js:63:19)"
```

Forced a 503 by turning off wifi
```
object
_msgid: "c8d62b5d.403e18"
topic: ""
payload: 1567619289863
error: object
message: "Error: getaddrinfo ENOTFOUND ***"
source: object
id: "fe9d378a.fdd1e8"
type: "influxdb in"
name: "Test Direct"
count: 1
stack: "Error: getaddrinfo ENOTFOUND *****
at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)"
```

Comment (malcolmwhat) [reply influxdb.js]: Do you see what I mean? Is there something I'm not understanding about how to use the first argument of node.error maybe? Maybe my logging is wrong too, I'm actually logging the msg. However when I added a function and attempted to access the `_error` not sure if it's on the global context or something, I would have thought it would be in the `msg`, I got a reference error

Comment (mblackstock) [reply influxdb.js]: yeah, you can't really mess with the error parameter passed in to node.error().  Looking at the node-red code, it pulls out just the fields it wants/expects and copies them over.  see https://github.com/node-red/node-red/blob/afaf077aca517b1de15a2e3e95277853a0b64549/packages/node_modules/%40node-red/runtime/lib/nodes/flows/Flow.js#L486


If you set msg.error, it looks like that code copies that to msg._error.  See https://github.com/node-red/node-red/blob/afaf077aca517b1de15a2e3e95277853a0b64549/packages/node_modules/%40node-red/runtime/lib/nodes/flows/Flow.js#L483-L485

I tried it yesterday, and seemed to work, but didn't test thoroughly with the different error types (ResultError and RequestError) thrown by the library.

Comment (mblackstock) [reply influxdb.js]: If we do this in each catch block:

```
msg.error = {
  statusCode: err.res ? err.res.statusCode : 503
};
node.error(err, msg);
```

it works.  A field called `_error` appears in the message along with the `error` field in the message to the catch node.  We can add more fields to this if needed later.

Comment (malcolmwhat) [reply influxdb.js]: Looks alright to me. The only gripe I might have is that you would have to track that down to the Node-Red runtime (as you've done here) to see that the field being injected into `error` is now in `_error`. Is this a documented node-red pattern (i.e. like recommended in the help or anything like that)? Maybe we could use another name like `influx_error` or something?

Comment (mblackstock) [reply influxdb.js]: I think we'd need to document it either way.  I suspect that magical move from `error` to `_error` is to support the edge case where a message already has an error field... so you're right, we probably don't want to depend on that.  Sure - lets put the fields in `influx-error`.  Do you want to make the change here and test it out, document it in the README?

Comment (malcolmwhat) [reply influxdb.js]: Ok so it's kind of frustrating that they don't throw the same error in the read and the write functions but I've updated the code to be the same for all of them. Basically the read only throws an Error whereas the writes (from the influx client) throw the connection information as well. Anyway this basically results in the reads always being a `503` if they fail, and the writes will have error codes like 404 for database not available, 400 for bad write, and 503 for a refused connection from the database.

My use-case is pretty much uniquely for writes, so this doesn't matter to me, but it might impact your decision on if you want to merge.

```
{ Error: Error from InfluxDB: database not found: testmeasurement
    at assertNoErrors (C:\Users\mw00120\bidsolutions\node-red\node_modules\influx\lib\src\results.js:92:19)
    at Object.parse (C:\Users\mw00120\bidsolutions\node-red\node_modules\influx\lib\src\results.js:108:5)
    at queryRaw.then.res (C:\Users\mw00120\bidsolutions\node-red\node_modules\influx\lib\src\index.js:849:68)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  message: 'Error from InfluxDB: database not found: testmeasurement' }
279
{ statusCode: 503 }
```

**Comment by malcolmwhat** [comment]
Alright, finally got around to it, let me know what you think.
